packages/@aws-cdk/aws-s3-assets/lib/asset.ts

```diff
@@ -103,6 +169,40 @@ export class Asset extends cdk.Construct implements assets.IAsset {
   constructor(scope: cdk.Construct, id: string, props: AssetProps) {
     super(scope, id);
 
+    // Bundling
+    let bundleAssetPath: string | undefined;
+    if (props.bundling) {
+      // Create the directory for the bundle next to the asset path
+      const bundlePath = path.join(path.dirname(props.path), props.bundling.bundleDirectory ?? '.bundle');
+      if (!fs.existsSync(bundlePath)) {
+        fs.mkdirSync(bundlePath);
+      }
+
+      bundleAssetPath = path.join(bundlePath, path.basename(props.path));
+      if (!fs.existsSync(bundleAssetPath)) {
+        fs.mkdirSync(bundleAssetPath);
+      }
+
+      const volumes = [
+        {
+          hostPath: props.path,
+          containerPath: BUNDLING_INPUT_DIR,
+        },
+        {
+          hostPath: bundleAssetPath,
+          containerPath: BUNDLING_OUTPUT_DIR,
+        },
+        ...props.bundling.volumes ?? [],
+      ];
+
+      props.bundling.image._run({
+        command: props.bundling.command,
+        volumes,
+        environment: props.bundling.environment,
+        workingDirectory: props.bundling.workingDirectory ?? BUNDLING_INPUT_DIR,
+      });
```

Comment:
Are error messages surfaced sufficiently to debug when the user hits an issue with bundling, both when some of the properties are incorrect and when the commands fail?

Reply:
This is handled in `_run()` but we can enclose the call to `run()` in a nicer `try/catch`.